Added progress bars to mission buttons

maxwellpark opened the pull request:
Intended to resemble those from the design doc.

Sliders reach 100% when mission completes and are reset when mission is restarted. Before, the sliders were getting reset at 1 tick before completion. This way the progress bars remain at 100% until clicked again, but this is preferable? Could add a coroutine or something but seems an ugly to do it.

maxwellpark committed:
Added sliders to mission buttons

Minor change to make the mission UI resemble the design doc

maxwellpark committed:
Added sliders (progress bars) to mission buttons

Sliders reach 100% when mission completes and are reset when mission is restarted. Before, the sliders were getting reset at 1 tick before completion. This way the progress bars remain at 100% until clicked again, but this is preferable? Could add a coroutine or something but seems an ugly to do it.

ambid17 commented on Packages/manifest.json:
```diff
@@ -1,11 +1,18 @@
-    "com.unity.ads": "2.0.8",
-    "com.unity.analytics": "3.3.2",
+    "com.unity.2d.sprite": "1.0.0",
+    "com.unity.2d.tilemap": "1.0.0",
```
uhhh whats with all the dependency changes

maxwellpark replied:
no idea, could it be to do with upgrading the project version? I think develop is still 2019.3.7f1.
I remember you upgraded it but that was on the messages-ui branch?

maxwellpark replied:
Either way, is it a big deal?

ambid17 replied:
lol not really, was just curious what happened

ambid17 commented on Assets/Scripts/MissionButton.cs:
```diff
@@ -16,7 +17,7 @@ public class MissionButton : MonoBehaviour
 
     public Mission mission;
-    public bool performingMission;
+    public bool performingMission;  
```
why add extra whitespace? 😂

ambid17 commented on Assets/Scripts/MissionButton.cs:
```diff
@@ -31,6 +32,7 @@ public void SetMissionTime(int timeLeftInSeconds)
     {
         TimeSpan missionDuration = new TimeSpan(0, 0, timeLeftInSeconds);
         countdownText.text = missionDuration.ToString("c");
+        this.slider.value += 1f / (float)this.mission.missionDurationInSeconds;
```
keyword "this" is unnecessary

ambid17 commented on Assets/Scripts/GameManager.cs:
```diff
@@ -61,6 +63,7 @@ IEnumerator WaitForMission(MissionButton missionButton)
         playerData.playerMoney += mission.missionValue;
         missionButton.ResetMissionTime();
         playerMoneyText.text = "$" + playerData.playerMoney;
+        mission.inProgress = false; 
```
it probably makes more sense to lower the  `yield return new WaitForSeconds(1);` to something like 0.1 seconds so the slider can move smoothly, in case there's a timer that goes less than a second once they have upgrades

maxwellpark committed:
Addressing requested changes. Not working yet, but the idea is there.

Periodic progress file has a dynamic scale factor defined in the MissionConstants class.

maxwellpark committed:
Scale factor determines how gradual the progress bars fill up.

maxwellpark committed:
Merge branch 'develop' into feature/mission-selection-ui
